api/server/router/container/container_routes.go

```diff
@@ -512,9 +513,29 @@ func (s *containerRouter) postContainersCreate(ctx context.Context, w http.Respo
 		}
 	}
 
+	if hostConfig != nil && versions.LessThan(version, "1.42") {
+		// Ignore BindOptions.CreateMountpoint because it was added in API 1.42.
+		for _, m := range hostConfig.Mounts {
+			if bo := m.BindOptions; bo != nil {
+				bo.CreateMountpoint = false
+			}
+		}
+	}
+
 	if hostConfig != nil && versions.GreaterThanOrEqualTo(version, "1.42") {
 		// Ignore KernelMemory removed in API 1.42.
 		hostConfig.KernelMemory = 0
+		for _, m := range hostConfig.Mounts {
+			if o := m.VolumeOptions; o != nil && m.Type != mount.TypeVolume {
+				return errdefs.InvalidParameter(fmt.Errorf("VolumeOptions must not be specified on mount type %q", m.Type))
+			}
```

Comment:
I'm still somewhat on the fence on moving these validations out of `validateMountConfigImpl`;

- previously, these checks where platform-specific (they were in `linuxParser.validateMountConfigImpl()`
- some checks were already performed on API versions before 1.42 (which were now removed)
- given, the checks were incomplete (so the discussion was; can we enable them for "all API versions") quite likely we _can_, but that could be a concern.
- `linuxParser.validateMountConfigImpl` is called through;
  - `linuxParser.parseMountSpec()` -> `linuxParser.ParseMountRaw()`
  - `linuxParser.parseMountSpec()` -> `linuxParser.ParseMountSpec()`

Both of those are part of various code-paths, which means that some of those paths won't perform a full validation; perhaps that's not a major issue, but it _could_ be (would have to think of possible scenarios).

Reply:
We can restore the checks in `validateMountConfigImpl` to limit impacts, will be checked twice worst case, which isn't a big deal. using 1.42 will just make them more strict

Reply:
What if we would reset the fields that were not validated in API < 1.42, so that validation won't invalidate them. That way we can keep all validation in `validateMountConfigImpl()`, but still keep the < 1.42 compatibility.

i.e.

- move the validation (back) in `validateMountConfigImpl()`
- change the loop at line 516 to also reset these (I'll post my suggestion there)